File: Stage8/Python/Problem1193.py
def main():
    X = int(input())
    line = 1
    count = 0
    
    while count + line < X:
        count += line
        line += 1
        
    index_line = X - count
    
    if line % 2 == 0:
        numerator = index_line
        denominator = line - index_line + 1
    else:
        numerator = line - index_line + 1
        denominator = index_line
    
    print(f"{numerator}/{denominator}")
    
if __name__ == "__main__":
    main()

File: Stage8/Java/Problem1193.java
package Stage8.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Problem1193 {
    public static void main(String[] args)throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int X = Integer.parseInt(br.readLine());

        int line = 1;
        int count = 0;

        while (count + line < X) {
            count += line;
            line++;
        }

        int indexLine = X - count;
        int numerator, denominator;
        
        if (line % 2 == 0) {
            numerator = indexLine;
            denominator = line - indexLine + 1;
        } else {
            numerator = line - indexLine + 1;
            denominator = indexLine;
        }

        System.out.println(numerator + "/" + denominator);
        br.close();
    }
}
